### PR TITLE
Maid 1103 rest api example

### PR DESCRIPTION
**Pull request by krishnaIndia**
Example implemented and root directory handling implemented in nfs. Blob creation bug fixed

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/maidsafe_client/60)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @ustulation

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: examples/rest_api.rs [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+                        return;
+                    } else {
+                        match containers.iter().find(|dir| *dir.get_name() == to_dir_name) {
+                            Some(info) => {
+                                match from_container.copy_blob(get_user_string("Blob name"), info.get_id()) {
+                                    Ok(_) => println!("Blob copied"),
+                                    Err(msg) => println!("Failed :: {}", msg)
+                                }
+                            },
+                            None => println!("Destination conatiner not found")
```
Comment (Viv-Rajkumar) [inline examples/rest_api.rs]: typo in error message

**Comment by Viv-Rajkumar** [comment]
Review status: 0 of 15 files reviewed, 3 unresolved discussions, all commit checks successful.

---

<sup>**[examples/rest_api.rs, line 374 \[r1\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrT1L1CRxlLDFceI6fi)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/771da07c521d9c018d2f374164c1e851c7f55e30/examples/rest_api.rs#L374)):</sup>
typo

---

<sup>**[src/nfs/rest/container.rs, line 87 \[r1\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrT1piI7mNL68IL0oYW)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/771da07c521d9c018d2f374164c1e851c7f55e30/src/nfs/rest/container.rs#L87)):</sup>
typo..prolly just find and replace "conatiner" to "container"

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60)

**Comment by krishnaIndia** [comment]
Review status: 0 of 15 files reviewed, 3 unresolved discussions, some commit checks pending.

---

<sup>**[examples/rest_api.rs, line 374 \[r1\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrT1L1CRxlLDFceI6fi)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/771da07c521d9c018d2f374164c1e851c7f55e30/examples/rest_api.rs#L374)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60)

**Comment by krishnaIndia** [comment]
Review status: 0 of 15 files reviewed, 3 unresolved discussions, some commit checks pending.

---

<sup>**[src/nfs/rest/container.rs, line 87 \[r1\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrT1piI7mNL68IL0oYW)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/771da07c521d9c018d2f374164c1e851c7f55e30/src/nfs/rest/container.rs#L87)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60)

**Comment by ustulation** [comment]
Review status: all files reviewed, 9 unresolved discussions, all commit checks successful.
Reviewed files:
- Cargo.toml @ r1
- examples/rest_api.rs @ r2
- README.md @ r1
- src/nfs/directory_info.rs @ r1
- src/nfs/directory_listing.rs @ r1
- src/nfs/file.rs @ r1
- src/nfs/helper/directory_helper.rs @ r1
- src/nfs/helper/file_helper.rs @ r1
- src/nfs/io/network_storage.rs @ r1
- src/nfs/io/reader.rs @ r1
- src/nfs/io/writer.rs @ r1
- src/nfs/metadata.rs @ r1
- src/nfs/mod.rs @ r1
- src/nfs/rest/blob.rs @ r1
- src/nfs/rest/container.rs @ r2

---

<sup>**[src/nfs/directory_info.rs, line 77 \[r2\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrWUOHx8P-Fab57Ay1g)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/2ed8f1e9748a71c0ac354579f25c5b01aac59ee4/src/nfs/directory_info.rs#L77)):</sup>
trailing space

---

<sup>**[src/nfs/directory_listing.rs, line 101 \[r2\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrWURsbGyl7PLv4mGZz)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/2ed8f1e9748a71c0ac354579f25c5b01aac59ee4/src/nfs/directory_listing.rs#L101)):</sup>
trailing space

---

<sup>**[src/nfs/helper/directory_helper.rs, line 97 \[r2\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrWU_DPBcQx46Rjw1C7)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/2ed8f1e9748a71c0ac354579f25c5b01aac59ee4/src/nfs/helper/directory_helper.rs#L97)):</sup>
is this not covered in the match statement below?

---

<sup>**[src/nfs/rest/container.rs, line 44 \[r2\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrWUz1iziEPo3jJTgEf)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/2ed8f1e9748a71c0ac354579f25c5b01aac59ee4/src/nfs/rest/container.rs#L44)):</sup>
why cloning an ARC when in the same thread .. `client.lock().unwrap().get_root_directroy_id().is_none()` should work.

---

<sup>**[src/nfs/rest/container.rs, line 53 \[r2\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrWVGX9buJBjKy81R1Y)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/2ed8f1e9748a71c0ac354579f25c5b01aac59ee4/src/nfs/rest/container.rs#L53)):</sup>
see the previous comment

---

<sup>**[src/nfs/rest/container.rs, line 61 \[r2\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrWVIuihUcWkwJ6uIVS)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/2ed8f1e9748a71c0ac354579f25c5b01aac59ee4/src/nfs/rest/container.rs#L61)):</sup>
see previous comment

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60)

**Comment by krishnaIndia** [comment]
Review status: all files reviewed, 9 unresolved discussions, all commit checks successful.

---

<sup>**[src/nfs/helper/directory_helper.rs, line 97 \[r2\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrWU_DPBcQx46Rjw1C7)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/2ed8f1e9748a71c0ac354579f25c5b01aac59ee4/src/nfs/helper/directory_helper.rs#L97)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60)

**Comment by krishnaIndia** [comment]
Review status: all files reviewed, 9 unresolved discussions, all commit checks successful.

---

<sup>**[src/nfs/rest/container.rs, line 44 \[r2\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrWUz1iziEPo3jJTgEf)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/2ed8f1e9748a71c0ac354579f25c5b01aac59ee4/src/nfs/rest/container.rs#L44)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60)

**Comment by krishnaIndia** [comment]
Review status: all files reviewed, 9 unresolved discussions, all commit checks successful.

---

<sup>**[src/nfs/rest/container.rs, line 53 \[r2\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrWVGX9buJBjKy81R1Y)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/2ed8f1e9748a71c0ac354579f25c5b01aac59ee4/src/nfs/rest/container.rs#L53)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60)

**Comment by krishnaIndia** [comment]
Review status: all files reviewed, 9 unresolved discussions, all commit checks successful.

---

<sup>**[src/nfs/rest/container.rs, line 61 \[r2\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrWVIuihUcWkwJ6uIVS)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/2ed8f1e9748a71c0ac354579f25c5b01aac59ee4/src/nfs/rest/container.rs#L61)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60)

**Comment by krishnaIndia** [comment]
Review status: all files reviewed, 9 unresolved discussions, all commit checks successful.

---

<sup>**[src/nfs/directory_info.rs, line 77 \[r2\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrWUOHx8P-Fab57Ay1g)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/2ed8f1e9748a71c0ac354579f25c5b01aac59ee4/src/nfs/directory_info.rs#L77)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60)

**Comment by krishnaIndia** [comment]
Review status: all files reviewed, 9 unresolved discussions, all commit checks successful.

---

<sup>**[src/nfs/directory_listing.rs, line 101 \[r2\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrWURsbGyl7PLv4mGZz)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/2ed8f1e9748a71c0ac354579f25c5b01aac59ee4/src/nfs/directory_listing.rs#L101)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60)

**Comment by krishnaIndia** [comment]
Review status: all files reviewed, 9 unresolved discussions, all commit checks successful.

---

<sup>**[examples/rest_api.rs, line 338 \[r1\]](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60#-JrSfLsLFN5juiw4MnmX-r1-338)** ([raw file](https://github.com/maidsafe/maidsafe_client/blob/771da07c521d9c018d2f374164c1e851c7f55e30/examples/rest_api.rs#L338)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/maidsafe_client/60)

**Comment by rossmuir** [comment]
test this please

**Comment by ustulation** [comment]
test this please

**Pull request merged**